docs/src/modules/utils/useCustomizationPlayground.tsx

```diff
@@ -0,0 +1,385 @@
+import * as React from 'react';
+import pick from 'lodash/pick';
+import { blue, pink } from '@mui/material/colors';
+import { BoxProps } from '@mui/material/Box';
+import { createTheme, ThemeProvider, styled, useTheme, Theme } from '@mui/material/styles';
+
+export type CustomizationLabelType = {
+  [k in 'customTheme' | 'styledComponents' | 'sxProp']: string;
+};
+
+type CustomizationItemType = {
+  type: 'warning' | 'success';
+  comments?: string;
+  componentProps?: Object;
+};
+export type CustomizationItemsType = Partial<{
+  [k in keyof CustomizationLabelType]: CustomizationItemType;
+}>;
+
+export type PickersSubcomponentType = {
+  [k: string]: { examples: CustomizationItemsType; slots: string[] };
+};
+
+export interface UseCustomizationPlaygroundProps {
+  examples: PickersSubcomponentType;
+  children?: React.ReactElement | React.ReactElement[];
+  componentName: string;
+}
+export const customizationLabels: CustomizationLabelType = {
+  customTheme: 'Custom theme',
+  styledComponents: 'Styled components',
+  sxProp: 'sx prop',
+};
+
+export const DEFAULT_COLORS = { pink, blue };
+export type ColorKey = keyof typeof DEFAULT_COLORS;
+
+export type StyleTokensType = {
+  color: ColorKey;
+  borderRadius: number;
+  borderWidth: number;
+};
+const styleTokens: StyleTokensType = {
+  color: 'blue',
+  borderRadius: 2,
+  borderWidth: 1,
+};
+export type HandleTokenChangeType = (
+  token: keyof Partial<typeof styleTokens>,
+  value: string | keyof typeof DEFAULT_COLORS | number,
+) => void;
+
+export type UseCustomizationPlaygroundReturnType = {
+  selectedDemo: string | null;
+  customizationOptions: Partial<CustomizationLabelType> | null;
+  selectedCustomizationOption: keyof CustomizationLabelType | null;
+  selectDemo: (interactionTarget: string | null) => void;
+  setSelectedCustomizationOption: (
+    customizationOption: keyof CustomizationLabelType | null,
+  ) => void;
+  selectedSlot: string | null;
+  setSelectedSlot: (slot: string | null) => void;
+  codeExample: string | null;
+  availableSlots: string[] | null;
+  handleTokenChange: HandleTokenChangeType;
+  selectedTokens: StyleTokensType;
+  selectedExample?: CustomizationItemType | null;
+};
+
+export function withStyles(
+  Component: React.ElementType,
+  selectedTokens: StyleTokensType,
+  selectedCustomizationOption: string | null,
+  selectedDemo: string | null,
+  selectedSlot: string | null,
+) {
+  return function StyledChild<T extends BoxProps>(props: T) {
+    const defaultTheme = useTheme();
+
+    const tokens = {
+      borderRadius: selectedTokens.borderRadius,
```

Comment:
As discussed with Nora, I'd personally be in favor of ditching the bordel related information to have a style as concise as possible and focus solely on how to pass some CSS to a piece of the UI (not how to write CSS)

From what I understood, this is not the vision of the design team.
I think we can settle on the current set of tokens and have a discussion in the coming weeks to decide the direction we would like to take for this component